specs/src/README.md

```diff
@@ -1,19 +0,0 @@
-# Celestia App Specifications
```

Comment:
Deleted this file b/c it's not necessary and a maintenance burden. It frequently becomes out of sync with SUMMARY.md which is the only thing that matters for generating the specs book